Makefile

```diff
@@ -285,7 +285,7 @@ stop-kind:
 	@$(MAKE) run CMD='-c "./build/local_kubernetes.sh stop_localkube"'
 
 check:
-	@./build/check.sh
+	@$(MAKE) run CMD='-c "./build/check.sh"'
```

Comment:
This ends up checking the Docker image, instead of the local setup. If I set `BUILD_IMAGE=debian:bookworm` in the `Makefile`, this step will fail. 

Can you try this change on OSX to see if it works? I changed the script to use `sh` shell and replace the upper case symbols with `awk`.

```sh
diff --git a/build/check.sh b/build/check.sh
index c6c78c2d..f9f034d6 100755
--- a/build/check.sh
+++ b/build/check.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 
 GO_BIN=go
 DOCKER_BIN=docker
@@ -11,12 +11,12 @@ failed="❌"
 pad=$(printf '%0.1s' "."{1..25})
 pad_len=25
 
-function output() {
-  local bin=$1
+output() {
+  local bin=$(echo "$1" | awk '{print toupper(substr( $0, 1, 1 )) substr($0, 2)}')
   local result=$2
-  local msg=$3
+  local msg=$(echo "$3" | awk '{print toupper(substr( $0, 1, 1 )) substr($0, 2)}')
 
-  printf "* need %s" "${bin^}"
+  printf "* need %s" "${bin}"
   printf "%*.*s %s" 0 $((pad_len - ${#bin})) "${pad}" "${result}"
   if [ ! -z "${msg}" ]; then
     printf "\n → %s" "${msg}"
@@ -24,33 +24,33 @@ function output() {
   printf "\n"
 }
 
-function check_version_go() {
+check_version_go() {
   local result=""
 
   if command -v ${GO_BIN} > /dev/null 2>&1 ; then
     result="${ok}"
   else
     result="${failed}"
-    msg="${GO_BIN^} not installed"
+    msg="${GO_BIN} not installed"
   fi
 
   output "${GO_BIN}" "${result}" "${msg}"
 }
 
 
-function check_version_docker() {
+check_version_docker() {
   local result=""
   if command -v ${DOCKER_BIN} > /dev/null 2>&1 ; then
     result="${ok}"
   else
     result="${failed}"
-    msg="${DOCKER_BIN^} not installed"
+    msg="${DOCKER_BIN} not installed"
   fi
 
   output "${DOCKER_BIN}" "${result}" "${msg}"
 }
 
-function check_version_kubectl() {
+check_version_kubectl() {
   local result=""
   if command -v ${KUBECTL_BIN} > /dev/null 2>&1 ; then
     result="${ok}"
@@ -59,7 +59,7 @@ function check_version_kubectl() {
     msg="${KUBECTL_BIN} not installed"
   fi
 
-  output "${KUBECTL_BIN^}" "${result}" "${msg}"
+  output "${KUBECTL_BIN}" "${result}" "${msg}"
 }
 
 check_version_go
```

Let me know if you have any questions.

Reply:
Yes. This step does fail after making these specific changes and running the check inside the Docker image. The issue was raised because the script fails on OSX as well.

Reply:
Can we try applying the above patch, and run the script on OSX, without using Docker? The purpose of `make check` is to validate the toolset on the developer's machine, not a Docker container.

Reply:
Awesome - feel free to update this PR with the patch. Then we can merge it.